plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -30,8 +33,27 @@ internal class LocationPuckManager(
 
   @VisibleForTesting(otherwise = PRIVATE)
   internal var lastLocation: Point? = null
+
+  @VisibleForTesting(otherwise = PRIVATE)
+  internal var lastMercatorScale = 1.0
+    set(value) {
+      if (abs(value - field) > MERCATOR_SCALE_THRESHOLD) {
+        field = value
+        (settings.locationPuck as? LocationPuck3D)?.let { locationPuck3D ->
+          locationLayerRenderer.styleScaling(get3DPuckScaleExpression(locationPuck3D, field))
+        }
+      }
+    }
+
   private val onLocationUpdated: ((Point) -> Unit) = {
     lastLocation = it
+    if (settings.locationPuck is LocationPuck3D) {
+      val latitude =
+        if (delegateProvider.mapProjectionDelegate.getMapProjection() == MapProjection.Globe) {
+          delegateProvider.mapCameraManagerDelegate.cameraState.center.latitude()
```

Comment:
pretty interesting that for globe we take not the actual puck location but camera center - why's that? imo it's more logical to take actual puck latitude all the time

Reply:
It's based on experimentation more details please check the internal channel, seems the camera capping only affects Globe projection. Need deeper look from native team to handle this appropriately.